packages/block-editor/src/components/list-view/branch.js

```diff
@@ -174,8 +176,14 @@ function ListViewBranch( props ) {
 						: `${ position }`;
 				const hasNestedBlocks = !! innerBlocks?.length;
 
+				const isZoomOutSectionBlock =
+					isZoomOutMode &&
+					sectionBlockClientIds?.includes( clientId );
+
 				const shouldExpand =
-					hasNestedBlocks && shouldShowInnerBlocks
+					! isZoomOutSectionBlock &&
+					hasNestedBlocks &&
+					shouldShowInnerBlocks
 						? expandedState[ clientId ] ?? isExpanded
 						: undefined;
```

Comment:
Instead of adding custom code to the inner parts of List View to filter out blocks, I think a better alternative might be to pass in a filtered block tree at the top of List View.

This approach is already used for filtering out disabled blocks. Some references:
- `useListViewClientIds`, this used to show all blocks, but was updated to use `getEnabledClientIdsTree` that instead only shows enabled blocks - https://github.com/WordPress/gutenberg/blob/07c7821b07e5d2b2abaa86f6737dd486da26383e/packages/block-editor/src/components/list-view/use-list-view-client-ids.js#L25-L26
- `getEnabledClientIdsTree`, this is the selector - https://github.com/WordPress/gutenberg/blob/07c7821b07e5d2b2abaa86f6737dd486da26383e/packages/block-editor/src/store/private-selectors.js#L82-L99

My thinking is you could add another private selector `getZoomedOutClientIdsTree` and then switch to using that in `useListViewClientIds` whenever zoomed out is active.

Reply:
Though I'm also not sure why `getEnabledClientIdsTree` is not enough. Does zoom out disable the inner block via some means that isn't `blockEditingMode`?

Maybe that's the thing to solve, along the lines of what @youknowriad mentioned here - https://github.com/WordPress/gutenberg/pull/65027#discussion_r1751429978 🤔

Reply:
Yeh I agree. Great feedback. I'll take a look at that approach.

Reply:
I took a look and worked out that we don't need to do anything as `getEnabledClientIdsTree` _already_ does what we want. I've updated the PR to reflect this.

The problem is that `getEnabledClientIdsTree` is memozied and doesn't re-compute unless we add `state.editorMode` to the dependencies array. This is what we discussed before.

Why is this? 

Surely switch to Zoom Out changes the blocks' `blockEditingModes` and as `state.blockEditingModes` is listed as a dependency of the `getEnabledClientIdsTree` `createSelector` it should recompute when editor mode changes to Zoom Out. Right?

Wrong.

In fact that assumption turns out to be incorrect. We never directly called `setBlockEditingMode` when switcing to Zoom Out mode. Instead the `getBlockEditingMode` _selector_ determines the correct `blockEditingMode` based on it's own internal logic rather than the state.

https://github.com/WordPress/gutenberg/blob/b96f5a659f0a6be6fe711b4a0c2d953f2a8d6254/packages/block-editor/src/store/selectors.js#L2910-L2925

This means that the `state.blockEditingModes` does not change which means that the `state.blockEditingModes` dependency does not cause `getEnabledClientIdsTree` to recompute. This in turn means the List View does not update to reflect the change.

Reply:
> Therefore I think the only logical course for now is to add the state.editorMode to the dependencies list of getEnabledClientIdsTree.

I think that makes sense. Thanks for digging into the issue!